opentuner/search/technique.py

```diff
@@ -204,15 +223,20 @@ def call_main_generator(self):
                                     self.pending_tests)
         if self.pending_tests:
           self.rounds_since_novel_request = 0
-          yield None # wait
+          yield False # wait
 
 #list of all techniques
 the_registry = list()
 
+#list of composable techniques
+the_composable_registry = list()
 
 def register(t):
   the_registry.append(t)
 
+def register_composable(t):
```

Comment:
My first thought is to have technique instances have some kind of "generate" method that copies itself, modifies the copy, then returns the copy. Then have a "register_generator" method (name subject to change) instead of the register_composable. Now if we want to randomly generate a technique, we'd just pick a random generator from the list of known generators and call "generate" to get a generated technique.

This has a bunch of limitations and tradeoffs, (e.g. what if sometimes we only want a specific kind of generation, we need instances of the techniques already etc.) but it's hard to consider these properly unless we more clearly define/specify/scope what the general case of "generating techniques" should entail (in terms of functionality and what the user should see/have to do)

I'll keep thinking of alternatives for now